core/src/ldml/C9134_ldml_markers.md

```diff
@@ -0,0 +1,73 @@
+# LDML Markers in Keyman
+
+## Background
+
+- This is the proposed implementation of markers for Keyman
+
+> Markers are placeholders which record some state, but without producing normal visible text output. They were designed particularly to support dead-keys.
+
+- See [tr35-keyboards-markers][] for source spec.
+    - [CLDR-16837][] has been filed to capture feedback on the marker spec, please add to that with any commnets.
+
+### Format
+
+- The general format is `\m{id}` where `id` is presumed to be `[0-9A-Za-z_]{1,32}`
+- `\m{.}` in matching matches a single marker.
+
+### Location
+
+Markers can appear in both 'emitting' and 'matching-only' areas:
+
+#### Emitting
+
+- `key to=` for emitting markers
+- `transform to=` to emit markers
+- `string value=` for matching or emitting markers
+
+#### Match only
+
+- `transform from=` to match markers
+- `transform after=` to match markers
+- `display to=` for matching keys which contain markers
+
+## Theory / Encoding
+
+- Keyman already uses U_SENTINEL `U+FFFF` (noncharacter)
+- The general proposal here is to use the sequence `U+FFFF U+EXXX` to represent marker #XXX
+- `U+FFFF` cannot otherwise occur in text, so it is unique
+- `U+EXXX` is always a private use character, so the marker number cannot collide with non-PUA text, however it may certainly collide with PUA text. The intention here is to reduce this possibility of collision, at least when (a) humans read a binary stream or (b) human _error_ causes the marker to show up in acutal text.  As a counter example, if we used `U+FFFF U+0022` to indicate marker 0x22, then the marker might show up as a doublequote (`"`).
```

Comment:
It's worth noting that the kmx keyboard processor uses `U+0001` and up as its marker code space. These have never escaped the confines of Keyman Core (and in fact, are entirely internal to kmx keyboard processor and are not even visible on any API surfaces, except debug log ones).

So, the proposed safety net may not be necessary -- if we start at `U+0001`, then we get 65533 (`U+0000`, `U+FFFF`, and `U+FFFE` are reserved) codes to play with, which I think is more generous than 4095 markers, even if we can't see an immediate need for that many.

Reply:
I think following in kmx's footsteps is an even better safety net